src/librustdoc/clean/types.rs

```diff
@@ -113,6 +113,50 @@ impl Item {
         self.attrs.doc_value()
     }
 
+    /// Convenience wrapper around [`Self::from_inner`] which converts `hir_id` to a [`DefId`]
+    pub fn from_hir_id_and_parts(
+        hir_id: hir::HirId,
+        name: Option<Symbol>,
+        inner: ItemEnum,
+        cx: &DocContext<'_>,
+    ) -> Item {
+        Item::from_def_id_and_parts(cx.tcx.hir().local_def_id(hir_id).to_def_id(), name, inner, cx)
+    }
+
+    pub fn from_def_id_and_parts(
+        def_id: DefId,
+        name: Option<Symbol>,
+        inner: ItemEnum,
+        cx: &DocContext<'_>,
+    ) -> Item {
+        use super::Clean;
+        use hir::Node;
+
+        debug!("name={:?}, def_id={:?}", name, def_id);
+
+        // `span_if_local()` lies about functions and only gives the span of the function signature
```

Comment:
Looks like you are re-implementing `fn span_with_body` here.

Reply:
I didn't know that existed, thanks!